src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/repository/CourseActivityRepository.kt

```diff
@@ -1,8 +1,24 @@
 package uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository
 
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
 import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
 import org.springframework.stereotype.Repository
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseActivity
 
 @Repository
-interface CourseActivityRepository : JpaRepository<CourseActivity, Long>
+interface CourseActivityRepository : JpaRepository<CourseActivity, Long> {
+
+  @Query(
+    value = """
+    select ca.courseActivityId from CourseActivity ca 
+    where ca.prison.id = :prisonId and ca.courseActivityId in 
+      (
+       select distinct(opp.courseActivity.courseActivityId) from OffenderProgramProfile opp
+       where opp.prison.id = :prisonId and opp.programStatus.code = 'ALLOC'
+      )   
+  """,
+  )
+  fun findActivitiesToMigrate(prisonId: String, pageRequest: Pageable): Page<Long>
+}
```

Comment:
Next up - adding more rules to the query

Reply:
I hate the syntax of specifications. I can read and understand SQL. So I'm going to try SQL first and switch if I really have to.